File: remove-bg.js
const sharp = require('sharp');
const path = require('path');

async function removeBackground() {
  const inputPath = path.join(__dirname, 'syncLabs logo.png');
  const outputPath = path.join(__dirname, 'syncLabs-logo-nobg.png');
  
  // Get raw pixels
  const { data, info } = await sharp(inputPath)
    .raw()
    .toBuffer({ resolveWithObject: true });
  
  const w = info.width, h = info.height;
  const channels = info.channels;
  
  // Sample background color (top-left corner)
  const bgR = data[0];
  const bgG = data[1];
  const bgB = data[2];
  console.log('Background color detected:', bgR, bgG, bgB);
  
  // New buffer with alpha channel
  const newData = Buffer.alloc(w * h * 4);
  
  // Process each pixel
  for (let y = 0; y < h; y++) {
    for (let x = 0; x < w; x++) {
      const srcIdx = (y * w + x) * channels;
      const dstIdx = (y * w + x) * 4;
      
      const r = data[srcIdx];
      const g = data[srcIdx + 1];
      const b = data[srcIdx + 2];
      
      // Copy RGB
      newData[dstIdx] = r;
      newData[dstIdx + 1] = g;
      newData[dstIdx + 2] = b;
      
      // Check if this pixel is background (dark color matching corners)
      // Background is around RGB(0, 10, 36) - make it transparent
      const isBg = (r < 20 && g < 30 && b < 60);
      
      // Also check for white/light background variations
      const isWhiteBg = (r > 240 && g > 240 && b > 240);
      
      newData[dstIdx + 3] = (isBg || isWhiteBg) ? 0 : 255;
    }
  }
  
  // Save as PNG with alpha
  await sharp(Buffer.from(newData), {
    raw: { width: w, height: h, channels: 4 }
  })
  .png()
  .toFile(outputPath);
  
  console.log('Saved:', outputPath);
  
  // Verify
  const outMeta = await sharp(outputPath).metadata();
  console.log('Output has alpha:', outMeta.hasAlpha);
}

removeBackground().catch(console.error);

File: remove_bg.py
from PIL import Image
import numpy as np

# Open the logo image
img = Image.open("syncLabs logo.png").convert("RGBA")

# Get the corner pixels (assuming top-left is the background color)
corner_color = img.getpixel((0, 0))

# Convert to numpy array for processing
data = np.array(img)

# Get the corner color components
r, g, b, a = corner_color

# Create a mask for pixels that match the corner color (within tolerance)
# Using a tolerance to handle slight color variations
tolerance = 30
mask = (
    (np.abs(data[:, :, 0] - r) <= tolerance) &
    (np.abs(data[:, :, 1] - g) <= tolerance) &
    (np.abs(data[:, :, 2] - b) <= tolerance)
)

# Set alpha to 0 for background pixels
data[:, :, 3] = np.where(mask, 0, data[:, :, 3])

# Create new image with transparent background
result = Image.fromarray(data, mode="RGBA")

# Save as PNG with transparency
result.save("public/syncLabs-logo.png", "PNG")

print("Background removed successfully!")
